src/app/script/tabs_widget.cpp

```diff
@@ -0,0 +1,166 @@
+// Aseprite
+// Copyright (c) 2022-2023  Igara Studio S.A.
+//
+// This program is distributed under the terms of
+// the End-User License Agreement for Aseprite.
+
+#include "app/script/tabs_widget.h"
+
+#ifdef ENABLE_UI
+
+using namespace ui;
+
+namespace app {
+namespace script {
+
+// static
+ui::WidgetType Tabs::Type()
+{
+  static ui::WidgetType type = ui::kGenericWidget;
+  if (type == ui::kGenericWidget)
+    type = ui::register_widget_type();
+  return type;
+}
+
+Tabs::Tabs(int selectorFlags) : m_selector(new HBox()),
+                                m_sepL(separator()),
+                                m_sepR(separator()),
+                                m_sepTB(separator()),
+                                m_buttonsBox(new HBox()),
+                                m_pages(new VBox()),
+                                m_selectorFlags(selectorFlags)
```

Comment:
You can avoid pointers of widgets if they are always created (using values like `HBox m_selector;`, etc.), e.g. https://github.com/aseprite/aseprite/blob/f4aa19ad8c12a97988383498dd39dd2dcc0e6d22/src/app/ui/color_bar.h#L208

Reply:
I thought in using as you suggested, but since it wasn't clear to me what happens when the delete operator is called on such values (or pointers to values) I decided to avoid it. Could you clarify this point?

Reply:
The `~Widget` destructor removes the widget itself from its parent, so this prevent a double deletion (as children will remove themselves from the parent). The order of execution when the Tabs is deleted is this:

1. ~Tabs is called
2. After that the destructor of each of its field is called (so ~HBox & ~Widget of each children is called, etc.)
3. Finally the ~Widget is called for the tabs widget

If a widget is still a children of Tabs (e.g. it was added with `addChild(new ...)`), it will be deleted automatically because ~Widget will delete the chlidren of the tabs widget.

Here an example: https://godbolt.org/z/rhGEdTfnr

Reply:
I see it now, thanks!